src/client/tile.h

```diff
@@ -233,7 +245,7 @@ struct TileSpec
 			material.MaterialType = video::EMT_TRANSPARENT_ALPHA_CHANNEL;
 			break;
 		case TILE_MATERIAL_LIQUID_TRANSPARENT:
-			material.MaterialType = video::EMT_TRANSPARENT_VERTEX_ALPHA;
+			material.MaterialType = video::EMT_TRANSPARENT_ALPHA_CHANNEL;
```

Comment:
Liquids currently use vertex alpha, which ignores the alpha of the texture (although the water texture in minetest_game is transparent, in fact `alpha=160` makes it transparent in the node definition).
What I do is to use the texture's alpha channel instead of the vertex's alpha (so the vertex can store the day/night difference). This is what `correctAlpha()` does: converts `alpha` to `^[opacity:` to store the opacity in the texture.